### DISCUSSION TITLE
Fixed number of tokens?

### DISCUSSION THREAD

**Discussion by EfficientCapital-Osterdahl** [vladimir-bukhtoyarov/bucket4j#166] [Ideas]
I am looking of an example for working with a fixed number of tokens. That is; I don't want to limit to a rate-per-minute; I want to limit to a certain number of concurrent requests. This would require that I also control the release the tokens. 

Is bucket4j the appropriate solution also in this scenario? 

My application is a REST service using Spring Boot.

**Comment by vladimir-bukhtoyarov** [comment]
Hello @fficientCapital-Osterdahl

It is not a use-case for Bucket4j. 

Use:
* ```java.util.concurrent.Semaphore``` if it is enough to control the number of parallel requests in the one JVM.
* [Hazelcast ISemaphore](https://docs.hazelcast.com/imdg/4.2/data-structures/isemaphore.html) or something comparable, if you need the limit that needs to be sharable across multiple JVMs.
